Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -2430,6 +2430,64 @@ bool Element::check_visibility(Optional<CheckVisibilityOptions> options)
     return true;
 }
 
+// https://drafts.csswg.org/css-contain/#proximity-to-the-viewport
+void Element::determine_proximity_to_the_viewport()
+{
+    // An element that has content-visibility: auto is in one of three states when it comes to its proximity to the viewport:
+
+    // - The element is close to the viewport: In this state, the element is considered "on-screen": its paint
+    //   containment box's overflow clip edge intersects with the viewport, or a user-agent defined margin around the
+    //   viewport.
+    auto viewport_rect = document().viewport_rect();
+    // NOTE: This margin is meant to allow the user agent to begin preparing for an element to be in the
+    // viewport soon. A margin of 50% is suggested as a reasonable default.
+    viewport_rect.inflate(viewport_rect.width(), viewport_rect.height());
```

Comment:
These want to be `/ 2` to add 50%, right? The spec doesn't seem to be that fussy but just wanted to check this isn't an accident.

Reply:
I had them as `/ 2` at first until I noticed: The spec wants a margin of 50%, which is half a viewport on all sides.
Inflate takes the total width/height to add, so if I `/ 2` them, each side on gets 25% of the viewport as a margin.

Reply:
Ah, that makes sense!